### PR TITLE
Adding a contributing page.

### PR DESCRIPTION
**Pull request by sfshaza2**
@Hixie, because of all the repo restructuring, this was moved to a new PR.

All feedback is incorporated.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+
```
Comment (Hixie) [inline CONTRIBUTING.md]: Looks like this lost the header, does that mean it's not an actual page on the Web site?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+in the [Flutter SDK](https://github.com/flutter/flutter) repo.
+
+We are also happy to accept your PRs to the
+[Flutter website](https://github.com/flutter/website/) repo.
+
+For ways to get involved in the Flutter community or to learn about us,
+visit the [Flutter community](/community) page.
+
+# Ways to contribute
+
```
Comment (Hixie) [inline CONTRIBUTING.md]: the "Ways to contribute" section doesn't include the ways that are mentioned in the introduction, such as fixing typos or sending PRs.

**Comment by Hixie** [comment]
Can you clarify the role of this page relative to the /community page and the CONTRIBUTING.md page on the flutter repo?

File: CONTRIBUTING.md (hunk 1 above)
Comment (sfshaza2) [reply CONTRIBUTING.md]: Correct. I moved it up one level.

File: CONTRIBUTING.md (hunk 2 above)
Comment (sfshaza2) [reply CONTRIBUTING.md]: Addressed.

**Comment by sfshaza2** [comment]
The CONTRIBUTING page is part of the repo, so doesn't need front matter.

The community page is being designed by an web designer. It will have pretty icons that link to different communities and sites. It will link to this page in the github repo.  There is a top-level link in the website banner that links to the community page.

**Comment by sfshaza2** [comment]
Please take another look, @Hixie

File: CONTRIBUTING.md (hunk 1 above)
Comment (Hixie) [reply CONTRIBUTING.md]: if it's not on the web site, who's the target audience? It's probably fine to just point to the flutter/flutter CONTRIBUTING.md doc, at that point. (I'm rewriting that one now, FWIW.)

**Comment by Hixie** [comment]
I think if the community page is the web site page then we're better off just having the community page point to the contributing doc on the main github repo.

**Comment by chalin** [comment]
FYI, we have distinct `CONTRIBUTING.md` files for AngularDart vs. the AngularDart website because the contributing guidelines are different:

- https://github.com/dart-lang/angular/blob/master/CONTRIBUTING.md
- https://github.com/dart-lang/site-webdev/blob/master/CONTRIBUTING.md

I believe that we should maintain a similar kind of separation.

**Comment by kwalrath** [comment]
Also, there's special support in GitHub for the top-level CONTRIBUTING.md file. Every repo should have one, even if it mostly just points to another page.

**Comment by sfshaza2** [comment]
I also feel we should have a CONTRIBUTING page for the website repo. We actually receive many PRs from the community, and the website is quite separate from the SDK. 

Unless you have problems with this page, @Hixie, can I merge it? I'd like to have it in place for the website launch, which is very soon.  Th

**Comment by sfshaza2** [comment]
OK, we lack consensus on the need for a contributing page, but the folks I have spoken to feel that every website needs a contributing page.

@Hixie, I am merging this for now, but we can open a bug, if you like, to review this decision.

**Comment by Robe21** [comment]
> @Hixie, because of all the repo restructuring, this was moved to a new PR.
> 
> All feedback is incorporated.

**Pull request merged**